scripts/generate_dump

```diff
@@ -25,15 +34,87 @@ DO_COMPRESS=true
 CMD_PREFIX=
 SINCE_DATE="@0" # default is set to January 1, 1970 at 00:00:00 GMT
 REFERENCE_FILE=/tmp/reference
+TECHSUPPORT_TIME_INFO=/tmp/techsupport_time_info
 BASE=sonic_dump_`hostname`_`date +%Y%m%d_%H%M%S`
 DUMPDIR=/var/dump
 TARDIR=$DUMPDIR/$BASE
 TARFILE=$DUMPDIR/$BASE.tar
 LOGDIR=$DUMPDIR/$BASE/dump
 NUM_ASICS=1
+HOME=${HOME:-/root}
+USER=${USER:-root}
+TIMEOUT_MIN="5"
+SKIP_BCMCMD=0
+
+save_bcmcmd() {
+    local start_t=$(date +%s%3N)
+    local end_t=0
+    local cmd="$1"
+    local filename=$2
+    local filepath="${LOGDIR}/$filename"
+    local do_gzip=${3:-false}
+    local tarpath="${BASE}/dump/$filename"
+    local timeout_cmd="timeout --foreground ${TIMEOUT_MIN}m"
+    [ ! -d $LOGDIR ] && $MKDIR $V -p $LOGDIR
 
+    if [ $SKIP_BCMCMD -eq 1 ]; then
+        echo "Skip $cmd"
+        return 0
+    fi
+    # eval required here to re-evaluate the $cmd properly at runtime
+    # This is required if $cmd has quoted strings that should be bunched
+    # as one argument, e.g. vtysh -c "COMMAND HERE" needs to have
+    # "COMMAND HERE" bunched together as 1 arg to vtysh -c
+    if $do_gzip
+    then
+        if $NOOP; then
+            echo "${timeout_cmd} $cmd &> '${filepath}'"
+        else
+            eval "${timeout_cmd} $cmd" &> "${filepath}"
+            ret=$?
+            if [ $ret -ne 0 ]; then
+                if [ $ret -eq 124 ]; then
+                    echo "Command: $cmd timedout after ${TIMEOUT_MIN} minutes."
+                else
+                    grep "polling socket timeout: Success" ${filepath} &>/dev/null
+                    if [ $? -eq 0 ]; then
+                        echo "bcmcmd command timeout. Setting SKIP_BCMCMD to true ..."
+                        SKIP_BCMCMD=1
+                    fi
+                fi
+            fi
+        fi
+        gzip ${filepath} 2>/dev/null
+        tarpath="${tarpath}.gz"
+        filepath="${filepath}.gz"
+    else
+        if $NOOP; then
+            echo "${timeout_cmd} $cmd &> '$filepath'"
+        else
+            eval "${timeout_cmd} $cmd" &> "$filepath"
+            ret=$?
+            if [ $ret -ne 0 ]; then
+                if [ $ret -eq 124 ]; then
+                    echo "Command: $cmd timedout after ${TIMEOUT_MIN} minutes."
+                else
+                    grep "polling socket timeout: Success" ${filepath} &>/dev/null
+                    if [ $? -eq 0 ]; then
+                        echo "bcmcmd command timeout. Setting SKIP_BCMCMD to true ..."
+                        SKIP_BCMCMD=1
+                    fi
+                fi
+            fi
+        fi
+    fi
```

Comment:
The only difference between the 'zip' and 'no zip' option seems to be the 3 statements. Can you refactor the code to reduce duplicate statements.

Reply:
how about:
```
if $do_gzip
        gzip ${filepath} 2>/dev/null
        tarpath="${tarpath}.gz"
        filepath="${filepath}.gz"
fi
```

Reply:
Done